subcommands/keys/ca_create.go

```diff
@@ -105,29 +128,53 @@ func doCreateCA(cmd *cobra.Command, args []string) {
 	writeFile("sign_tls_csr", *resp.SignTlsScript, 0700)
 
 	fmt.Println("Creating offline root CA for Factory")
-	run("./create_ca")
+
+	var factoryKey *ecdsa.PrivateKey
+	var factoryCa *x509.Certificate
+	if useBashScripts {
+		run("./create_ca")
+	} else {
+		factoryKey, factoryCa = subcommands.CreateFactoryCa(factory)
+	}
 
 	fmt.Println("Signing Foundries TLS CSR")
-	resp.TlsCrt = run("./sign_tls_csr", "tls-csr")
+	if useBashScripts {
+		resp.TlsCrt = run("./sign_tls_csr", "tls-csr")
+	} else {
+		resp.TlsCrt = subcommands.SignTlsCsr([]byte(resp.TlsCsr), factoryCa, factoryKey)
+	}
```

Comment:
It'd be nice if this could be implemented via the build constraints.

What I mean is to have this layout directory:
```
x509/
    bash.go
    windows.go
```

Above, both files will declare a corresponding build constraint and implement a common interface:
```
// bash.go:
// go:build !windows

func CreateFactoryCa() {
    run("./create_ca")
}

// other functions below
```

```
// windows.go
// go:build windows

// This uses a global context to hold the CA operations intermediate results
// Even better approach would be if Windows implementation writes files to disk, just like Bash implementation.
// The user needs a backup of those files, as they contain important bits, just IMHO.

type context struct {
    rootCaKey *ecdsa.PrivateKey
    rootCaCert *x509.Certificate
}

var ctx context

func CreateFactoryCa() {
   ctx.rootCaKey = ...
   ctx.rootCaCert = ...
}

// other functions below
```

Then, the el2g and keys could simply call the common function interfaces, and be sure that a correct flavor is used.
For example, here you could write:
```
    // add above the import for github.com/foundriesio/fioctl/x509
    x509.CreateFactoryCa()
```

This way, we get a different binary for windows vs non-windows executables, which only contain the necessary bits.